crates/aztec_backend/src/barretenberg_rs/composer.rs

```diff
@@ -186,7 +186,7 @@ impl SchnorrConstraint {
 }
 #[derive(Clone, Hash, Debug)]
 pub struct MerkleMembershipConstraint {
-    pub hash_path: Vec<(i32, i32)>,
+    pub hash_path: Vec<i32>,
```

Comment:
For prosperity, can you detail why the hashpath layout is being changed?

Reply:
I moved away from the tuple hashpath layout originally originally as we were no longer using the DB which returned both the current node and neighbor node that were to be hashed together at each level. This removed the need to have to call `flatten_path` when we were solving the GadgetCall for merkle membership. Then in our discussion with @guipublic we then decided to not allow the possibility of overwriting the leaf node (or any other complementary node at future levels of the tree) as we generated the merkle root and checked for membership. This removed needing both hash path nodes in our constraints. A Noir developer is now passing in a vector with one node at each level. The adjacent node that used to also be included is now calculated by the backend.

Reply:
Keeping the old hashpath layout in the serialiser with the changes made would most likely require inserting the calculated intermediate nodes to the witness and the outputs of the gadget call. Which seems excessive as the inputs to the gadget call are now a vector of just one intermediate node at each level of the tree.

Reply:
missed this sorry